### PR TITLE
Add measurements toolbox

### PR DESCRIPTION
**Pull request by jo-mueller**
A step towards #80 . New features:

- Toolbox widget: I created a barebones measurement widget (type: QToolBox) with the three main tabs `Curvature`, `Shape spectrum` and `Geometry`. The current code only provides measurements for the `Curvature` tab - also only one out of the 6 (!) measurements for curvature of the original code: The averaged mean curvature (e.g., `curvature.mean()`)
- Demo notebook and interactive tutorial for example data

A few notes:

- I designed the toolbox widget in a way so that plotting will be done lazily later-on, e.g. so that plots are only updated if they are currently on display. 
- The analysis toolbox currently only handles 3D data and no 4D points data. However, when using workflows, this is not so much of an issue since the workflow only ever shows a single frame at a time. Hence, it is only necessary to connect the time slider to the update function of the toolbox :) Lateron, we should however make sure that the toolbox handles 4D data equally well.
Let me know what you think @haesleinhuepf

### CODE REVIEW

**Comment by jo-mueller** [comment]
Also: There is a failing test, but it's not the test that's failing rather than the teardown of a qt instance - I'm afraid I don't really know how to fix this one :S

File: docs/tutorials/point_and_click/demo_analyze_spherical_harmonics_expansion.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+## Analyze the approximated surface
+
+napari-stress provides a number of quantitative parameters for a spherical harmonics surface expansion. To access these measurements, select the respective plugin from the tools menu (`Tools > Measurement > Surface curvature from points (n-STRESS)`). This plugin performs a spherical harmonics expansion with the parameters described in the [respective tutorial](./demo_spherical_harmonics.md). Additionally, the plugin calculates so-called `quadrature points`, which allow for fast and accurate measurement of surface characteristics. For more information on these points (valid values, etc), check out the [notebook](../demo/demo_analyze_spherical_harmonics.ipynb). 
+
+Make sure to check the `run analysis toolbox` checkbox to proceed to the analysis toolbox:
+
+<img src="./imgs/demo_analyze_spherical_harmonics1.png" width="45%">
+
+The result is a toolbox plugin showing the histogram of the calculated curvature and the averaged mean curvature:
```
Comment (haesleinhuepf) [inline docs/tutorials/point_and_click/demo_analyze_spherical_harmonics_expansion.md:15]: As mentioned in the code, I don't think it is a good idea to have checkboxes opening another user-interface. There are better ways for implementing this. I'm happy to brainstorm together with you in front of a white-board.

File: docs/tutorials/point_and_click/demo_analyze_spherical_harmonics_expansion.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# Analyzing a spherical harmonics expansion
+
+After approximating a pointcloud with a spherical harmonics expansion, we may want to extract quantitative features (e.g., curvature, surface area, etc.). The stress-code provides a multitude of quantitative features for this purpose, which shall be demonstrated in this document. To get started, create a pointcloud according to the workflow suggestions in this repository or load the sample data from napari-stress (`File > Open Sample > napari-stress: Dropplet pointcloud`).
```
Comment (haesleinhuepf) [inline docs/tutorials/point_and_click/demo_analyze_spherical_harmonics_expansion.md:3]:
```suggestion
This tutorial describes how quantitative features such as curvature, surface area, etc. can be determined after approximating a pointcloud with a spherical harmonics expansion. To get started, create a pointcloud as explained in [this tutorial](../workflows/Surface_to_curvature.md) or load the sample data from the menu `File > Open Sample > napari-stress: Droplet pointcloud`.
```
If the menu still is called "Dropplet", please rename it to "Droplet".

**Review by haesleinhuepf — CHANGES_REQUESTED** [review]
Hi Johannes @jo-mueller ,

great work! It looks like you're working towards guiding the user more and to connect workflow-steps better. I suggest to not build in check-boxes so that the next user interface opens. If we did this more often, code would become messy. Instead, we should think about 
a) How to connect processing-steps with a generic user-interface with "Previous" and "Next" buttons, like in TrackMate for example, or
b) Build a "super"-user-interface as proposed in #80 

Happy to discuss details in a meeting.

Best,
Robert

**Pull request closed without merging**